templates/project/cf.yml

```diff
@@ -0,0 +1,116 @@
+AWSTemplateFormatVersion: '2010-09-09' {{$accounts := .Accounts}} {{$project := .Project}} {{$apps := .Apps}}
+# Cross-regional resources deployed via a stackset in the tools account
+# to support the CodePipeline for a workspace
+Description: Cross-regional resources to support the CodePipeline for a workspace
+Metadata:
+  Version: {{.Version}}
+  Apps: {{if not $apps}}[]{{else}}{{range $app := $apps}}
+  - {{$app}}{{end}}{{end}}
+  Accounts: {{if not $accounts}}[]{{else}}{{range $account := $accounts}}
+  - {{$account}}{{end}}{{end}}
+Resources:
+  KMSKey:
+    # Used by the CodePipeline in the tools account to en/decrypt the
+    # artifacts between stages
+    Type: AWS::KMS::Key
+    Properties:
+      EnableKeyRotation: true
+      KeyPolicy:
+        Version: "2012-10-17"
+        Id: !Ref AWS::StackName
+        Statement:
+          -
+            # Allows the key to be administered in the tools account
+            Effect: Allow
+            Principal:
+              AWS: !Sub arn:aws:iam::${AWS::AccountId}:root
+            Action:
+              - "kms:Create*"
+              - "kms:Describe*"
+              - "kms:Enable*"
+              - "kms:List*"
+              - "kms:Put*"
+              - "kms:Update*"
+              - "kms:Revoke*"
+              - "kms:Disable*"
+              - "kms:Get*"
+              - "kms:Delete*"
+              - "kms:ScheduleKeyDeletion"
+              - "kms:CancelKeyDeletion"
+            Resource: "*"
+          -
+            # Allow use of the key in the tools account and all environment accounts
+            Effect: Allow
+            Principal:
+              AWS:
+                - !Sub arn:aws:iam::${AWS::AccountId}:root
+                {{range $accounts}}
+                - arn:aws:iam::{{.}}:root{{end}}
+            Action:
+              - kms:Encrypt
+              - kms:Decrypt
+              - kms:ReEncrypt*
+              - kms:GenerateDataKey*
+              - kms:DescribeKey
+            Resource: "*"
+  PipelineBuiltArtifactBucketPolicy:
+    Type: AWS::S3::BucketPolicy
+    DependsOn: PipelineBuiltArtifactBucket
+    Properties:
+      Bucket: !Ref PipelineBuiltArtifactBucket
+      PolicyDocument:
+        Statement:
+          -
+            Action:
+              - s3:*
+            Effect: Allow
+            Resource:
+              - !Sub arn:aws:s3:::${PipelineBuiltArtifactBucket}
+              - !Sub arn:aws:s3:::${PipelineBuiltArtifactBucket}/*
+            Principal:
+              AWS:
+                - !Sub arn:aws:iam::${AWS::AccountId}:root
+                {{range $accounts}}
+                - arn:aws:iam::{{.}}:root{{end}}
+  PipelineBuiltArtifactBucket:
+    Type: AWS::S3::Bucket
+    Properties:
+      VersioningConfiguration:
+        Status: Enabled
+    DeletionPolicy: Retain
+
+{{range $app := $apps}}
+  ECRRepo{{$app}}:
```

Comment:
The app name can contain dashes which will mess the logical id up :|

We might need to implement a macro to drop dashes

Reply:
Ah yea... I think I realized that at some point, but forgot to address it. I can convert it to camel case.

Reply:
I have a nice follow up PR to fix this. It's truly delightful.